File: snapshots/output-net10.0/syntax/VBMain/GlobalAttributes.vb
  Imports System.Diagnostics.CodeAnalysis
'         ^^^^^^ read_access scip-dotnet nuget . . System/
'                enclosing_range 0:8-0:26
'         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ import scip-dotnet nuget . . CodeAnalysis/
'                                         enclosing_range 0:8-0:39
'                ^^^^^^^^^^^ read_access scip-dotnet nuget . . Diagnostics/
'                            enclosing_range 0:8-0:26
'                            ^^^^^^^^^^^^ read_access scip-dotnet nuget . . CodeAnalysis/
'                                         enclosing_range 0:8-0:39

  Namespace VBMain
'           ^^^^^^ read_access scip-dotnet nuget . . VBMain/
'                  enclosing_range 2:10-2:16
      <SuppressMessage("ReSharper", "all")>
'      ^^^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 CodeAnalysis/SuppressMessageAttribute#`.ctor`().
'                      enclosing_range 3:5-3:20
      <AttributeUsage(AttributeTargets.[Class], AllowMultiple:=True, Inherited:=True)>
'      ^^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/AttributeUsageAttribute#`.ctor`().
'                     enclosing_range 4:5-4:19
'                     ^^^^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/AttributeTargets#
'                                      enclosing_range 4:20-4:44
'                                      ^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/AttributeTargets#Class.
'                                              enclosing_range 4:20-4:44
'                                               ^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/AttributeUsageAttribute#AllowMultiple.
'                                                             enclosing_range 4:46-4:59
'                                                                    ^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/AttributeUsageAttribute#Inherited.
'                                                                              enclosing_range 4:67-4:76
      Public Class GlobalAttributes
'                  ^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/GlobalAttributes#
'                                   enclosing_range 3:4-5:33
'                                   documentation ```vb\nClass GlobalAttributes\n```
'                                   kind Class
'                                   relationship implementation scip-dotnet nuget System.Runtime 10.0.0.0 System/Attribute#
          Inherits Attribute
'                  ^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/Attribute#
'                            enclosing_range 6:17-6:26

          Class AuthorAttribute
'               ^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/GlobalAttributes#AuthorAttribute#
'                               enclosing_range 8:8-8:29
'                               documentation ```vb\nClass AuthorAttribute\n```
'                               kind Class
'                               relationship implementation scip-dotnet nuget System.Runtime 10.0.0.0 System/Attribute#
              Inherits Attribute
'                      ^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/Attribute#
'                                enclosing_range 9:21-9:30

              Public Sub New(ByVal name As String)
'                        ^^^ definition scip-dotnet nuget . . VBMain/GlobalAttributes#AuthorAttribute#`.ctor`().
'                            enclosing_range 11:12-11:48
'                            documentation ```vb\nPublic Sub AuthorAttribute.New(name As String)\n```
'                            kind Constructor
'                                  ^^^^ definition scip-dotnet nuget . . VBMain/GlobalAttributes#AuthorAttribute#`.ctor`().(name)
'                                       enclosing_range 11:12-11:48
'                                       documentation ```vb\nname As String\n```
'                                       kind Parameter
'                                       relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/String#
              End Sub
          End Class

          <Author("PropertyAttribute")>
'          ^^^^^^ read_access scip-dotnet nuget . . VBMain/GlobalAttributes#AuthorAttribute#`.ctor`().
'                 enclosing_range 15:9-15:15
          Public Z As Integer
'                ^ definition scip-dotnet nuget . . VBMain/GlobalAttributes#Z.
'                  enclosing_range 15:8-16:27
'                  documentation ```vb\nPublic GlobalAttributes.Z As Integer\n```
'                  kind Field
'                  relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int32#

          <Author("MethodAttribute")>
'          ^^^^^^ read_access scip-dotnet nuget . . VBMain/GlobalAttributes#AuthorAttribute#`.ctor`().
'                 enclosing_range 18:9-18:15
          Private Function Method1() As Integer
'                          ^^^^^^^ definition scip-dotnet nuget . . VBMain/GlobalAttributes#Method1().
'                                  enclosing_range 18:8-19:45
'                                  documentation ```vb\nPrivate Function GlobalAttributes.Method1() As Integer\n```
'                                  kind Method
              Return 0
          End Function

          <Author("EnumAttribute")>
'          ^^^^^^ read_access scip-dotnet nuget . . VBMain/GlobalAttributes#AuthorAttribute#`.ctor`().
'                 enclosing_range 23:9-23:15
          Enum A
'              ^ definition scip-dotnet nuget . . VBMain/GlobalAttributes#A#
'                enclosing_range 23:8-24:14
'                documentation ```vb\nEnum A\n```
'                kind Enum
'                relationship implementation scip-dotnet nuget System.Runtime 10.0.0.0 System/IComparable#
'                relationship implementation scip-dotnet nuget System.Runtime 10.0.0.0 System/IConvertible#
'                relationship implementation scip-dotnet nuget System.Runtime 10.0.0.0 System/ISpanFormattable#
'                relationship implementation scip-dotnet nuget System.Runtime 10.0.0.0 System/IFormattable#
              B
'             ^ definition scip-dotnet nuget . . VBMain/GlobalAttributes#A#B.
'               enclosing_range 25:12-25:13
'               documentation ```vb\nA.B = 0\n```
'               kind Constant
'               relationship type_definition scip-dotnet nuget . . VBMain/GlobalAttributes#A#
              C
'             ^ definition scip-dotnet nuget . . VBMain/GlobalAttributes#A#C.
'               enclosing_range 26:12-26:13
'               documentation ```vb\nA.C = 1\n```
'               kind Constant
'               relationship type_definition scip-dotnet nuget . . VBMain/GlobalAttributes#A#
          End Enum

          <Author("EventAttribute")>
'          ^^^^^^ read_access scip-dotnet nuget . . VBMain/GlobalAttributes#AuthorAttribute#`.ctor`().
'                 enclosing_range 29:9-29:15
          Public Event SomeEvent As EventHandler
'                      ^^^^^^^^^ definition scip-dotnet nuget . . VBMain/GlobalAttributes#SomeEvent#
'                                enclosing_range 29:8-30:46
'                                documentation ```vb\nPublic Event GlobalAttributes.SomeEvent As EventHandler\n```
'                                kind Event
'                                relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/EventHandler#
'                                   ^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/EventHandler#
'                                                enclosing_range 30:34-30:46

          <Author("TypeParameterAttribute")>
'          ^^^^^^ read_access scip-dotnet nuget . . VBMain/GlobalAttributes#AuthorAttribute#`.ctor`().
'                 enclosing_range 32:9-32:15
          Public Class InnerClass(Of T)
'                      ^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/GlobalAttributes#InnerClass#
'                                 enclosing_range 32:8-33:37
'                                 documentation ```vb\nClass InnerClass(Of T)\n```
'                                 kind Class
'                                    ^ definition local 0
'                                      enclosing_range 32:8-33:37
'                                      documentation ```vb\nT\n```
'                                      kind TypeParameter
              Private Sub Method(Of T2)()
'                         ^^^^^^ definition scip-dotnet nuget . . VBMain/GlobalAttributes#InnerClass#Method().
'                                enclosing_range 34:12-34:39
'                                documentation ```vb\nPrivate Sub InnerClass(Of T).Method(Of T2)()\n```
'                                kind Method
'                                   ^^ definition local 1
'                                      enclosing_range 34:12-34:39
'                                      documentation ```vb\nT2\n```
'                                      kind TypeParameter
              End Sub
          End Class
      End Class
  End Namespace

File: snapshots/output-net10.0/syntax/Main/GlobalAttributes.cs
  using System.Diagnostics.CodeAnalysis;
//      ^^^^^^ read_access scip-dotnet nuget . . System/
//             enclosing_range 0:6-0:24
//      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ import scip-dotnet nuget . . CodeAnalysis/
//                                      enclosing_range 0:6-0:37
//             ^^^^^^^^^^^ read_access scip-dotnet nuget . . Diagnostics/
//                         enclosing_range 0:6-0:24
//                         ^^^^^^^^^^^^ read_access scip-dotnet nuget . . CodeAnalysis/
//                                      enclosing_range 0:6-0:37

  namespace Main;
//          ^^^^ read_access scip-dotnet nuget . . Main/
//               enclosing_range 2:10-2:14

  [SuppressMessage("ReSharper", "all")]
// ^^^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 CodeAnalysis/SuppressMessageAttribute#`.ctor`().
//                 enclosing_range 4:1-4:16
  [AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = true)]
// ^^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/AttributeUsageAttribute#`.ctor`().
//                enclosing_range 5:1-5:15
//                ^^^^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/AttributeTargets#
//                                 enclosing_range 5:16-5:38
//                                 ^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/AttributeTargets#Class.
//                                       enclosing_range 5:16-5:38
//                                        ^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/AttributeUsageAttribute#AllowMultiple.
//                                                      enclosing_range 5:40-5:53
//                                                              ^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/AttributeUsageAttribute#Inherited.
//                                                                        enclosing_range 5:62-5:71
  public class GlobalAttributes : Attribute
//             ^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/GlobalAttributes#
//                              enclosing_range 4:0-44:1
//                              documentation ```cs\nclass GlobalAttributes\n```
//                              kind Class
//                              relationship implementation scip-dotnet nuget System.Runtime 10.0.0.0 System/Attribute#
//                                ^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/Attribute#
//                                          enclosing_range 6:32-6:41
  {
      class AuthorAttribute : Attribute
//          ^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/GlobalAttributes#AuthorAttribute#
//                          enclosing_range 8:4-13:5
//                          documentation ```cs\nclass AuthorAttribute\n```
//                          kind Class
//                          relationship implementation scip-dotnet nuget System.Runtime 10.0.0.0 System/Attribute#
//                            ^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/Attribute#
//                                      enclosing_range 8:28-8:37
      {
          public AuthorAttribute(string name)
//               ^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/GlobalAttributes#AuthorAttribute#`.ctor`().
//                               enclosing_range 10:8-12:9
//                               documentation ```cs\npublic AuthorAttribute.AuthorAttribute(string name)\n```
//                               kind Constructor
//                                      ^^^^ definition scip-dotnet nuget . . Main/GlobalAttributes#AuthorAttribute#`.ctor`().(name)
//                                           enclosing_range 10:8-12:9
//                                           documentation ```cs\nstring name\n```
//                                           kind Parameter
//                                           relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/String#
          {
          }
      }

      [Author("PropertyAttribute")] public int Z;
//     ^^^^^^ read_access scip-dotnet nuget . . Main/GlobalAttributes#AuthorAttribute#`.ctor`().
//            enclosing_range 15:5-15:11
//                                             ^ definition scip-dotnet nuget . . Main/GlobalAttributes#Z.
//                                               enclosing_range 15:4-15:47
//                                               documentation ```cs\npublic int GlobalAttributes.Z\n```
//                                               kind Field
//                                               relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int32#

      [Author("MethodAttribute")]
//     ^^^^^^ read_access scip-dotnet nuget . . Main/GlobalAttributes#AuthorAttribute#`.ctor`().
//            enclosing_range 17:5-17:11
      int Method1()
//        ^^^^^^^ definition scip-dotnet nuget . . Main/GlobalAttributes#Method1().
//                enclosing_range 17:4-21:5
//                documentation ```cs\nprivate int GlobalAttributes.Method1()\n```
//                kind Method
      {
          return 0;
      }

      [Author("EnumAttribute")]
//     ^^^^^^ read_access scip-dotnet nuget . . Main/GlobalAttributes#AuthorAttribute#`.ctor`().
//            enclosing_range 23:5-23:11
      enum A
//         ^ definition scip-dotnet nuget . . Main/GlobalAttributes#A#
//           enclosing_range 23:4-28:5
//           documentation ```cs\nenum A\n```
//           kind Enum
//           relationship implementation scip-dotnet nuget System.Runtime 10.0.0.0 System/IComparable#
//           relationship implementation scip-dotnet nuget System.Runtime 10.0.0.0 System/IConvertible#
//           relationship implementation scip-dotnet nuget System.Runtime 10.0.0.0 System/ISpanFormattable#
//           relationship implementation scip-dotnet nuget System.Runtime 10.0.0.0 System/IFormattable#
      {
          B,
//        ^ definition scip-dotnet nuget . . Main/GlobalAttributes#A#B.
//          enclosing_range 26:8-26:9
//          documentation ```cs\nA.B = 0\n```
//          kind Constant
//          relationship type_definition scip-dotnet nuget . . Main/GlobalAttributes#A#
          C
//        ^ definition scip-dotnet nuget . . Main/GlobalAttributes#A#C.
//          enclosing_range 27:8-27:9
//          documentation ```cs\nA.C = 1\n```
//          kind Constant
//          relationship type_definition scip-dotnet nuget . . Main/GlobalAttributes#A#
      }

      [Author("EventAttribute")]
//     ^^^^^^ read_access scip-dotnet nuget . . Main/GlobalAttributes#AuthorAttribute#`.ctor`().
//            enclosing_range 30:5-30:11
      public event EventHandler SomeEvent
//                 ^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/EventHandler#
//                              enclosing_range 31:17-31:29
//                              ^^^^^^^^^ definition scip-dotnet nuget . . Main/GlobalAttributes#SomeEvent#
//                                        enclosing_range 30:4-35:5
//                                        documentation ```cs\npublic event EventHandler GlobalAttributes.SomeEvent\n```
//                                        kind Event
//                                        relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/EventHandler#
      {
          add { }
          remove { }
      }

      [Author("TypeParameterAttribute")]
//     ^^^^^^ read_access scip-dotnet nuget . . Main/GlobalAttributes#AuthorAttribute#`.ctor`().
//            enclosing_range 37:5-37:11
      public class InnerClass<[Author("ClassTypeParameter")] T>
//                 ^^^^^^^^^^ definition scip-dotnet nuget . . Main/GlobalAttributes#InnerClass#
//                            enclosing_range 37:4-43:5
//                            documentation ```cs\nclass InnerClass<T>\n```
//                            kind Class
//                             ^^^^^^ read_access scip-dotnet nuget . . Main/GlobalAttributes#AuthorAttribute#`.ctor`().
//                                    enclosing_range 38:29-38:35
//                                                           ^ definition local 0
//                                                             enclosing_range 37:4-43:5
//                                                             documentation ```cs\nT\n```
//                                                             kind TypeParameter
      {
          void Method<[Author("MethodTypeParameter")] T2>()
//             ^^^^^^ definition scip-dotnet nuget . . Main/GlobalAttributes#InnerClass#Method().
//                    enclosing_range 40:8-42:9
//                    documentation ```cs\nprivate void InnerClass<T>.Method<T2>()\n```
//                    kind Method
//                     ^^^^^^ read_access scip-dotnet nuget . . Main/GlobalAttributes#AuthorAttribute#`.ctor`().
//                            enclosing_range 40:21-40:27
//                                                    ^^ definition local 1
//                                                       enclosing_range 40:8-42:9
//                                                       documentation ```cs\nT2\n```
//                                                       kind TypeParameter
          {
          }
      }
  }
